types/src/consensus/attestation_metadata.rs

```diff
@@ -9,3 +9,27 @@ pub struct AttestationMetadata {
     pub global_sequence: u64,
     pub contract_sequences: Vec<(Address, u64)>,
 }
+impl Hashable for AttestationMetadata {
+    fn hash_custom(&self) -> Hash {
+        use alloy_primitives::Keccak256;
+
+        let mut hasher = Keccak256::default();
+        hasher.update(self.timestamp.hash_custom().as_slice());
+        hasher.update(self.global_sequence.to_be_bytes());
+
+        //sort to ensure deterministic, order-independent hashing
+        let mut sorted = self.contract_sequences.clone();
+        sorted.sort_by(|a, b| match a.0.cmp(&b.0) {
+            std::cmp::Ordering::Equal => a.1.cmp(&b.1),
+            other => other,
+        });
+
+        hasher.update((sorted.len() as u64).to_be_bytes());
```

Comment:
Is there any reason why big-endian bytes are preferred? It might be faster to use little-endian (most machines are LE); I'm not 100% sure.